packages/dynamite/dynamite/lib/src/models/type_result/some_of.dart

```diff
@@ -0,0 +1,97 @@
+part of 'type_result.dart';
+
+@immutable
+abstract class TypeResultSomeOf extends TypeResult {
+  TypeResultSomeOf(
+    super.className, {
+    required this.subTypes,
+    super.nullable,
+  }) : super(isTypeDef: false);
+
+  final Set<TypeResult> subTypes;
+
+  @override
+  String? get _builderFactory => null;
+
+  @override
+  String? get _serializer {
+    if (optimizedSubTypes.length <= 1) {
+      return null;
+    }
+
+    return '..add(${className}Extension.serializer)';
+  }
+
+  late final List<TypeResult> optimizedSubTypes = subTypes.toList();
+
+  @override
+  TypeResult get dartType {
+    if (optimizedSubTypes.length == 1) {
+      return optimizedSubTypes.single;
+    }
+
+    final record = optimizedSubTypes.map((final type) {
+      final dartType = type.nullableName;
+      final dartName = toDartName(dartType);
+      final fieldName = toFieldName(dartName, type.className);
+
+      return '$dartType $fieldName';
```

Comment:
This could be set to final, no?

Reply:
what do you mean?
`return 'final $dartType $fieldName';`? This will not work as it would result in a record of `({final int? $int, final String string})` which isn't valid dart syntax.

Reply:
Oh I thought this was possible 🫣